src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,305 @@
+
+  // Throw error if the image is incompatible with the matrix.
+  if (maxWidth > 0 && maxHeight > 0 &&
+     (width != maxWidth || height != maxHeight))
+  {
+    std::ostringstream oss;
+    oss << "Image '" << fileName;
+    oss << "' does not match matrix height or width." << std::endl;
+    oss << "Image width: " << width;
+    oss << ", Image height: " << height << std::endl;
+    oss << "Max Width: " << maxWidth;
+    oss << ", Max Height: " << maxHeight << std::endl;
+    throw std::runtime_error(oss.str());
+  }
+  return status;
+}
+
+bool Image::Load(const std::vector<std::string>& files,
+                 bool flipVertical,
+                 arma::Mat<unsigned char>&& outputMatrix)
+{
+  if (files.size() == 0)
+  {
+    std::ostringstream oss;
+    oss << "File vector is empty." << std::endl;
+    throw std::runtime_error(oss.str());
+  }
+
+  arma::Mat<unsigned char> img;
+  size_t width, height;
+  bool status = Load(files[0], flipVertical, std::move(img),
+                     width, height, channels);
+  Log::Info << "Loaded " << files[0] << std::endl;
+
+  // Decide matrix dimension using the image height and width.
+  maxWidth = std::max(maxWidth, width);
+  maxHeight = std::max(maxHeight, height);
+
+  outputMatrix.set_size(maxWidth*maxHeight*channels, files.size());
+  outputMatrix.col(0) = img;
+
+  for (size_t i = 1; i < files.size() ; i++)
+  {
+    status &= Load(files[i],
+                   flipVertical,
+                   std::move(outputMatrix.col(i)));
+    Log::Info << "Loaded " << files[i] << std::endl;
+  }
+  return status;
+}
+
+
+bool Image::Save(const std::string& fileName,
+                 size_t width,
+                 size_t height,
+                 size_t channels,
+                 bool flipVertical,
+                 arma::Mat<unsigned char>&& inputMatrix)
+{
+  if (!ImageFormatSupported(fileName, true))
+  {
+    std::ostringstream oss;
+    oss << "File type " << Extension(fileName) << " not supported.\n";
+    oss << "Currently it supports ";
+    for (auto extension : saveFileTypes)
+      oss << ", " << extension;
+    oss << std::endl;
+    throw std::runtime_error(oss.str());
+    return false;
+  }
+  if (inputMatrix.n_cols > 1)
+  {
+    std::cout << "Input Matrix contains more than 1 image." << std::endl;
+    std::cout << "Only the firstimage will be saved!" << std::endl;
+  }
+  unsigned char *image = inputMatrix.memptr();
+  stbi_flip_vertically_on_write(flipVertical);
+
+  bool status;
+  int tempWidth = width, tempHeight = height, tempChannels = channels;
+  if ("png" == Extension(fileName))
+  {
+    status = stbi_write_png(fileName.c_str(), tempWidth, tempHeight,
+        tempChannels, image, width*channels);
+  }
+  else if ("bmp" == Extension(fileName))
+  {
+    status = stbi_write_bmp(fileName.c_str(), tempWidth, tempHeight,
+        tempChannels, image);
+  }
+  else if ("tga" == Extension(fileName))
+  {
+    status = stbi_write_tga(fileName.c_str(), tempWidth, tempHeight,
```

Comment:
Not sure, but I thought the compiler will convert the type automatically? Or maybe we should just use good old c-type casting.

Reply:
To be on the safe side. However, it shows me an error, I guess that's because my compile is very old.

Reply:
Same if you use ``status = stbi_write_tga(fileName.c_str(), (int) tempWidth, (int) tempHeight, ...)``?

Reply:
Should I also add 

> Just saw your last two commit messages, for further commits it would be helpful if you could use a more descriptive commit message, makes it easier to search through the history and see right away what might have changed.

Apologies for that.

Reply:
no typecasting from size_t to int was giving an error . Similarly for char to string.